src/csharp/Microsoft.Spark/RDD/Collector.cs

```diff
@@ -66,15 +66,13 @@ internal interface IDeserializer
         }
 
         /// <summary>
-        /// Deserializer using the BinaryFormatter.
+        /// Deserializer using MessagePack.
         /// </summary>
         private sealed class BinaryDeserializer : IDeserializer
         {
-            private readonly BinaryFormatter _formater = new BinaryFormatter();
-
             public object Deserialize(Stream stream, int length)
             {
-                return _formater.Deserialize(stream);
+                return MessagePackSerializer.Typeless.Deserialize(stream);
             }
```

Comment:
There's a bug in MessagePack I found while tried to reuse these changes, when called from Collector::Collect: Deserializer erases entire stream after the first read. 

Workaround for it is to either read stream in small chunks and pass to MessagePack, or load entire stream in MemoryStream and then use it.


https://github.com/MessagePack-CSharp/MessagePack-CSharp/issues/1881